src/Nethermind/Nethermind.Core/ChainLevelInfo.cs

```diff
@@ -83,14 +83,17 @@ public void InsertBlockInfo(Keccak hash, BlockInfo blockInfo, bool setAsMain)
             BlockInfo[] blockInfos = BlockInfos;
 
             int? foundIndex = FindIndex(hash);
-            if (!foundIndex.HasValue)
+            if (foundIndex is null)
             {
                 Array.Resize(ref blockInfos, blockInfos.Length + 1);
             }
             else
             {
                 if (blockInfo.IsBeaconInfo && blockInfos[foundIndex.Value].IsBeaconMainChain)
                     blockInfo.Metadata |= BlockMetadata.BeaconMainChain;
+
+                if (blockInfo.EqualsIgnoringWasProcessed(blockInfos[foundIndex.Value]))
```

Comment:
Not sure if we need to check `EqualsIgnoringWasProcessed` at all. Hash should identify block.

Reply:
You mean use `BlockInfo.BlockHash`? But what about `TotalDifficulty` and `Metadata`? those fields are independent of the `BlockHash`:

https://github.com/NethermindEth/nethermind/blob/fe2b14e89a6b07767b460a0f509a5437af881093/src/Nethermind/Nethermind.Core/BlockInfo.cs#L23-L28